src/atomic_ptr.hpp

```diff
@@ -280,11 +286,9 @@ struct atomic_value_t
 #endif
 
 #if defined ZMQ_ATOMIC_PTR_MUTEX
-    mutex_t sync;
+    mutable mutex_t sync;
```

Comment:
Why has this become necessary?

Reply:
There's a const function somewhere in atomic_ptr.cpp that uses the lock/unlock functions of `sync`, and the implementation of mutex_t I made for VxWorks relies on VxWorks semaphores instead of POSIX mutexes; apparently the lock/unlock function changes the mutex_t object with VxWorks semaphores. I can make it a declaration only for VxWorks.

Reply:
I think it's fine to have the mutex mutable, this is a recurring pattern. I just could not relate it to a change that causes it to be necessary.